### ISSUE TITLE
Test time reported to JUnit runner is incorrect

### ISSUE THREAD

**Issue by xiaodongw** [kotest/kotest#1160]
I have the [gradle-test-logger](https://github.com/radarsh/gradle-test-logger-plugin) plugin in my build, it reports the slow tests if it takes more than 1 second to run.

This is the output for one of my test with JUnit5, this is an integration test, so it's kind of slow.
```
leap.filesystem.metadata.FsMetadataImplTest
    ✔ testCannotLockAgain() (2.1s)
    ✔ testPutVersionsWithEmptyList()
    ✔ testCannotCreateDuplicatedItemMetadata()
    ✔ testAcl()
    ✔ testItem()
    ✔ testRoot()
    ✔ testLockSuccessfully()
    ✔ testCannotCreateDuplicatedVersion()
    ✔ testPutItemMetadataWithEmptyList()
    ✔ testPutItemsWithEmptyList()
    ✔ testVersion()
    ✔ testLockCrossRoots() (2.1s)
    ✔ testNoDeadLock()
    ✔ Put same block twice will silently succeed
    ✔ testItemMetadata()
    ✔ testBlocks()
    ✔ testCommit()
``` 

After I converted this test to kotlintest, this the new output, you can see here the test time reported is incorrect, most of them have the same test time.
```
leap.filesystem.metadata.FsMetadataImplSpec
    ✔ root (4.2s)
    ✔ put/read should work (4.2s)
    ✔ put items with empty list (4.2s)
    ✔ has metadata (4.2s)
    ✔ can put empty item metadata (4.2s)
    ✔ cannot create duplicated item metadata (4.2s)
    ✔ commit (4.2s)
    ✔ create / read version (4.2s)
    ✔ can put empty versions (4.2s)
    ✔ cannot create duplicated version (4.2s)
    ✔ put / get (4.2s)
    ✔ put same block twice will silently succeed (4.2s)
    ✔ should work correctly (4.2s)
    ✔ can lock cross roots (4.2s)
    ✔ no deadlock (2.1s)
    ✔ cannot lock when locked (2.1s)
    ✔ should work
```

After dig a little bit in the kotlintest code, I found the problem is here in io.kotlintest.runner.junit5.JUnitTestRunnerListener, seems to support multiple invocations, it doesn't report the test result to JUnit after the test case is done, but only when the whole spec is completed.
```kotlin
override fun exitTestCase(testCase: TestCase, result: TestResult) {
    logger.trace("exitTestCase ${testCase.description} with result $result")
    // we don't immediately finish a test, we just store the result until we have completed the spec
    // this allows us to handle multiple invocations of the same test case, deferring the notification
    // to junit until all invocations have completed
    results.add(ResultState(testCase, result))
  }
```

I'm wondering if we can still report the correct test time, 1) by counting the runs of a test case, once it reaches the TestCaseConfig.invocations, report the result in it's last exitTestCase, or 2) at least provide the correct test time when invocations is set to 1.

**Comment by sksamuel** [comment]
Invocations are removed in 4.0. They are problematic. Instead you can just repeat the code inside the test.
So the timing in 4.0 is exactly as it should be.

**Comment by sksamuel** [comment]
I am closing this because I have confirmed it is fixed in 4.0 snapshot.

![image](https://user-images.githubusercontent.com/743706/72665410-d69e5100-39cd-11ea-9cf3-eef2107b6cca.png)
